lib/client/profile_e2e_test.go

```diff
@@ -0,0 +1,240 @@
+			profileList, outError = c.Profiles().List(meta1)
+			log.Println("Get profile list returns: ", profileList.Items)
+
+			// Get a profile with meta1.
+			outProfile1, outError1 = c.Profiles().Get(meta1)
+
+			// Assert they are equal and no errors.
+			Expect(profileList.Items[0].Spec).To(Equal(outProfile1.Spec))
+			Expect(outError1).To(BeNil())
+
+			By("Delete, Get and assert error")
+
+			// Delete a profile with meta1.
+			outError1 = c.Profiles().Delete(meta1)
+
+			// Get a profile with meta1.
+			_, outError = c.Profiles().Get(meta1)
+
+			// Expect an error since the profile was deleted.
+			Expect(outError.Error()).To(Equal(errors.New("resource does not exist: ProfileTags(name=profile1)").Error()))
+
+			// Delete the second profile with meta2.
+			outError1 = c.Profiles().Delete(meta2)
+
+			By("Delete all the profiles, Get profile list and expect empty profile list")
+
+			// Both profiles are deleted in the calls above.
+			// Get the list of all the profiles.
+			profileList, outError = c.Profiles().List(api.ProfileMetadata{})
+			log.Println("Get profile list returns: ", profileList.Items)
+
+			// Create an empty profile list.
+			// Note: you can't use make([]api.Profile, 0) because it creates an empty underlying struct,
+			// whereas new([]api.Profile) just returns a pointer without creating an empty struct.
+			emptyProfileList := new([]api.Profile)
+
+			// Expect returned profileList to contain empty profileList.
+			Expect(profileList.Items).To(Equal(*emptyProfileList))
+
+		},
+
+		// Test 1: Pass two fully populated ProfileSpecs and expect the series of operations to succeed.
+		Entry("Two fully populated ProfileSpecs",
+			api.ProfileMetadata{Name: "profile1"},
+			api.ProfileMetadata{Name: "profile2"},
+			*createAPIProfileSpecObject("profile1", []string{"profile1-tag1", "profile1-tag2"}),
+			*createAPIProfileSpecObject("profile2", []string{"profile2-tag1", "profile2-tag2"}),
+		),
+
+		// Test 2: Pass one fully populated ProfileSpec and another empty ProfileSpec and expect the series of operations to succeed.
+		Entry("One fully populated ProfileSpec and another empty ProfileSpec",
+			api.ProfileMetadata{Name: "profile1"},
+			api.ProfileMetadata{Name: "profile2"},
+			*createAPIProfileSpecObject("profile1", []string{"profile1-tag1", "profile1-tag2"}),
+			api.ProfileSpec{},
+		),
+
+		// Test 3: Pass one partially populated ProfileSpec and another fully populated ProfileSpec and expect the series of operations to succeed.
+		Entry("One partially populated ProfileSpec and another fully populated ProfileSpec",
+			api.ProfileMetadata{Name: "profile1"},
+			api.ProfileMetadata{Name: "profile2"},
+			api.ProfileSpec{
+				Tags: []string{"profile1-tag1"},
+			},
+			*createAPIProfileSpecObject("profile2", []string{"profile2-tag1", "profile2-tag2"}),
+		),
+	)
+})
+
+// profileCreate takes client and api.Profile and creates a profile.
+func profileCreate(c *client.Client, p *api.Profile) (*api.Profile, error) {
```

Comment:
Do we need these functions?  It seems they just add a trace - but shouldn't we be expecting the return values in the actual test anyway - in which case we'd get the output we want from the test.